### DISCUSSION TITLE
Output field numbering and firsttimetep in visnek

### DISCUSSION THREAD

**Discussion by timofeymukha** [ExtremeFLOW/neko#497] [General]
Killed quite a bit of time on the following. Having generated some solution data in `field` files, I was using the Nek5000 utility `visnek` to create a `.nek5000` file to open in Paraview. However, this would not work, both Paraview and Visit threw an error. Finally, I realized that the issue was that the first output is `field0.f00000`, but Paraview expected it to start with 1, i.e. `field0.f00001`. This can be controlled by setting `firsttimestep: 0` in the generated `.nek5000`  file.

Can one somehow make this work right automatically? Mabe one can pass firsttimestep as a parameter to visnek? Or should we just create visneko, which defaults this to 0? I guess, that anyway we should not rely on a nek5000 utility.

**Comment by njansson** [comment]
Hmm, didn't the `.nek5000` file that Neko spits out work directly?

**Reply by timofeymukha** [reply]
I did not realize neko creates its own .nek5000 :). It was a productive 2 hours today.

**Comment by pschlatt1** [comment]
Halfways related: the first file written out when setting nsamples is not at the initial time. Would it be useful to have a flag to write out the intial field as well, e.g. to check the initial conditions from user_ic? This would then mean that one perhaps should reserve field0.f00000 for that, and start the normal numbering at field0.f00001. to field0.f0nsamples.

**Reply by timofeymukha** [reply]
Sounds like a good idea.

**Reply by njansson** [reply]
Not sure, but isn't this what PR https://github.com/ExtremeFLOW/neko/pull/644 adds?
